### ISSUE TITLE
Compilation Error with `--combined-json` and `--via-ir` Options 

### ISSUE THREAD

**Issue by desp4c1to** [ethereum/solidity#14686]
#### Issue Description:

I'm developing a builder to compile multiple smart contracts. During this process, I've encountered a specific issue with the Solidity compiler (`solc`) when using certain command-line options.

#### Environment:
- Compiler: Solidity (`solc`)
- Command used: `solc xxx --optimize-runs=1 --combined-json abi,ast,bin,bin-runtime,srcmap,srcmap-runtime,userdoc,devdoc,hashes`

#### Steps to Reproduce:
1. Compile smart contracts using `solc` without any additional options. This step works without any issues.
2. Add the `--combined-json abi,ast,bin,bin-runtime,srcmap,srcmap-runtime,userdoc,devdoc,hashes` option to the `solc` command. This results in a `Stack too deep` error.
3. Follow the error suggestion and add the `--via-ir` option. However, this doesn't resolve the issue, and the compilation still fails.

#### Expected Behavior:
The compiler should successfully compile the smart contracts with the specified options without returning a stack error.

#### Actual Behavior:
When the `--combined-json` option is included, the compiler returns an error:
```
Error: Stack too deep. Try compiling with `--via-ir`...
```

Adding `--via-ir` as suggested results in the following error:
```
Uncaught exception:
[solidity::util::tag_comment*] = Variable var_amount_3216 is 1 slot(s) too deep inside the stack. Stack too deep. Try compiling with `--via-ir`...
```

Still results in the following error
```
Uncaught exception:
/Users/distiller/project/libyul/backends/evm/EVMObjectCompiler.cpp(126): Throw in function run
Dynamic exception type: boost::wrapexcept<solidity::yul::StackTooDeepError>
std::exception::what: Variable var_amount_3216 is 1 slot(s) too deep inside the stack. Stack too deep. Try compiling with `--via-ir` (cli) or the equivalent `viaIR: true` (standard JSON) while enabling the optimizer. Otherwise, try removing local variables.
[solidity::util::tag_comment*] = Variable var_amount_3216 is 1 slot(s) too deep inside the stack. Stack too deep. Try compiling with `--via-ir` (cli) or the equivalent `viaIR: true` (standard JSON) while enabling the optimizer. Otherwise, try removing local variables.
```

#### Additional Context:
- The issue seems to be specifically related to the combination of `--combined-json` and `--via-ir` options.
- I am compiling the contract from [this address](https://etherscan.io/address/0x9ab6b21cdf116f611110b048987e58894786c244#code) 

Any insights or suggestions on how to resolve this issue would be greatly appreciated.

**Comment by r0qs** [comment]
I could compile just fine your example using the via IR pipeline and the latest solc version on the `develop` branch (`0.8.24+commit.58811f13`), i.e. `solc --via-ir --optimize-runs=1 --combined-json abi,ast,bin,bin-runtime,srcmap,srcmap-runtime,userdoc,devdoc,hashes c.sol`

What version of `solc` are you using?

**Comment by desp4c1to** [comment]
I am using v0.8.19, the same as illustrate in the etherscan
```
solc, the solidity compiler commandline interface
Version: 0.8.19+commit.7dd6d404.Darwin.appleclang
```

**Comment by desp4c1to** [comment]
I can compile with 0.8.23
```
solc, the solidity compiler commandline interface
Version: 0.8.23+commit.f704f362.Darwin.appleclang
```

**Comment by nikola-matic** [comment]
> I can compile with 0.8.23
> 
> ```
> solc, the solidity compiler commandline interface
> Version: 0.8.23+commit.f704f362.Darwin.appleclang
> ```

This was most likely fixed in [0.8.21](https://github.com/ethereum/solidity/blob/develop/Changelog.md?plain=1#L72) by the default minimal yul optimizations (stack to memory mover), and since it can't be reproduced in 0.8.23, I'll be closing the issue as fixed.
